### ISSUE TITLE
srcset Proxying doesn't work with React

### ISSUE THREAD

**Issue by helgenlechner** [DevExpress/testcafe-hammerhead#2958]
### What is your Scenario?

I'm trying to load responsive images with srcsets in an integration test for a React app.

### What is the Current behavior?

Images cannot be loaded, because the proxying does not work correctly. The proxied URL is malformed.

### What is the Expected behavior?

Images with srcsets can be loaded and rendered correctly.

### What is your public website URL? (or attach your complete example)

Here's a minimal Sandbox where the behavior can be observed: https://codesandbox.io/s/srcset-proxying-test-58w6s7

The first image is rendered with a simple React app, that just renders the image with a srcset. The second image is just rendered in static HTML but has the same properties as the first image. In the preview, the image is visible twice, as expected: 
<img width="822" alt="Screenshot 2023-08-25 at 15 21 55" src="https://github.com/DevExpress/testcafe-hammerhead/assets/7873316/3f844297-f8d4-4420-877e-d929b803cbae">

When running the provided Testcafe script and test, the image rendered through React fails to load, and only the statically rendered image shows up: 
<img width="817" alt="Screenshot 2023-08-25 at 15 22 21" src="https://github.com/DevExpress/testcafe-hammerhead/assets/7873316/bfa52ec2-9e34-4e51-b993-3e08428ad54e">
(The image itself is different in the two screenshots because I'm using an sample image API.)

### What is your TestCafe test code?

Test Runner Script: https://codesandbox.io/s/srcset-proxying-test-58w6s7?file=/test/runTest.js

Test Script: https://codesandbox.io/s/srcset-proxying-test-58w6s7?file=/test/imgTest.js

### Your complete configuration file

None other than what is specified in the Sandbox.

### Steps to Reproduce

1. Run the Testcafe script that is provided in the Sandbox. I'm not sure if that is possible within the Sandox. You may have to paste the `runTest.js` and `imgTest.js` to local files.
2. The test will open the Sandbox and pause, showing one failed and one successfully loaded image.

### TestCafe version

2.6.2

### Node.js version

16.14.2

### Command-line arguments

node ./test/runTest.js

### Browser name(s) and version(s)

Chrome 116.0.5845.96

### Platform(s) and version(s)

macOS 13.4.1

### Other

I can't use Testcafe's new Native Automation because we run the tests in CI in Browserstack and in Chrome, Edge, Firefox and Safari.

**Comment by Aleksey28** [comment]
Hi @helgenlechner,

I didn't manage to reproduce the issue with the latest TestCafe version. Please update your TestCafe package.

**Comment by helgenlechner** [comment]
Hi @Aleksey28,

Thank you for taking a look. I have now updated to the latest version of TestCafe (3.3.0) and the issue still persists. I wonder how you got it to work? Are you using Native Automation? Like I stated in the original description, I can't use Native Automation, because we use remote browsers and also we need to test in Firefox and Safari.

<img width="1392" alt="Screenshot 2023-09-01 at 09 05 46" src="https://github.com/DevExpress/testcafe-hammerhead/assets/7873316/6ec73340-0133-4345-a514-00ad0919880e">
